### PR TITLE
Interpret scalar transformations as diagonal

### PR DESCRIPTION
**Pull request by NeilGirdhar**

### CODE REVIEW

File: lineax/_operator.py [hunk 1]
```diff
@@ -1514,7 +1514,11 @@ def is_diagonal(operator: AbstractLinearOperator) -> bool:
 @is_diagonal.register(JacobianLinearOperator)
 @is_diagonal.register(FunctionLinearOperator)
 def _(operator):
-    return diagonal_tag in operator.tags
+    return (
+        diagonal_tag in operator.tags
+        or operator.in_size() == 1
+        and operator.out_size() == 1
```
Comment (patrick-kidger) [inline lineax/_operator.py:1520]: I think it'd be good practice to group these in brackets: `diagonal_tag in operator.tags or (operator.in_size() == 1 and operator.out_size() == 1)`

I don't have the precedence rules for `x or y and z` in my head!

**Review by patrick-kidger — COMMENTED** [review]
This LGTM! Thank you for getting the tridiagonal one too, I hadn't noticed that a tridiagonal operator with length-zero off-diagonals should also be possible.

Would you be able to add some tests for this edge case?

File: lineax/_operator.py (hunk 1 above)
Comment (NeilGirdhar) [reply lineax/_operator.py]: Done!

**Comment by NeilGirdhar** [comment]
> Would you be able to add some tests for this edge case?

Done

**Comment by patrick-kidger** [comment]
LGTM - merged! :)
